glue/viewers/image/composite_array.py

```diff
@@ -80,7 +101,34 @@ def __call__(self, bounds=None):
         img = None
         visible_layers = 0
 
-        for uuid in sorted(self.layers, key=lambda x: self.layers[x]['zorder']):
+        # We first check that layers are either all colormaps or all single
+        # colors, and at the same time, if we use colormaps we check what
+        # the smallest alpha value is - if it is 1 then colormaps do not change
+        # transparency and this allows us to speed things up a little.
+
+        minimum_cmap_alpha = 1
+        if self.mode == 'colormap':
+            for layer in self.layers.values():
+                minimum_cmap_alpha = min(layer['cmap'](CMAP_SAMPLING)[:, 3].min(),
+                                            minimum_cmap_alpha)
+
+        # Get a sorted list of UUIDs with the top layers last
+        sorted_uuids = sorted(self.layers, key=lambda x: self.layers[x]['zorder'])
+
+        # In the case where we are dealing with colormaps, we can start from
+        # the last layer that has an opacity of 1 because layers below will not
+        # affect the output, assuming also that the colormaps do not change the
+        # alpha
+        if self.mode == 'colormap' and minimum_cmap_alpha == 1.:
```

Comment:
```suggestion
        if self.mode == 'colormap' and minimum_cmap_alpha == 1.:
            sorted_uuids = sorted_uuids[-1:]
        elif self.mode == 'colormap':
```
Not sure if I understand the combination of layers right, but if _all_ of them have alpha=1, wouldn't you automatically see only the last, topmost one? Whereas for the case that _some_ are transparent, still everything below the last opaque one could be ignored?
The first case could probably be caught even earlier by just picking the layer with highest `self.layers[x]['zorder']`, avoiding further creating of temporary arrays.

Reply:
@astrofrog I've added a test with a faded colormap and the use of layer-minimum alpha above – not sure if it will make a relevant difference, but I can push the commit if you want to review.

Reply:
> Still wondering if it would not suffice for the last layer of opacity 1, if that layer has minimum cmap_alpha of 1, to ignore all layers below.

Is that not effectively what we are doing now?

Reply:
If any layer has a cmap_alpha < 1 (`minimum_cmap_alpha ` < 1), we're alway keeping the full `sorted_uuids` list. Instead testing for `if layer['alpha'] * layer['cmap'](CMAP_SAMPLING)[:, 3].min() == 1` we may still truncate it at that layer (which works at least for my two-layer test).

Reply:
It _might_ actually work with computing `plane` further down as well (does at least with that same test), but that may need some more testing.

Reply:
Feel free to push the commit so I can make sure I understand :)

Reply:
I've pushed the first change to just set `sorted_uuids[start:]` separately so they are easier to revert.
This is still not really testing cmap alpha values _between_ 0 and 1.